### PR TITLE
Enhance notify

### PR DESCRIPTION
**Pull request by yccheok**
Sorry. I'm not sure what's wrong with my newly added code and unit test, which causes NPE. Do you have any idea?

    java.lang.NullPointerException
        at android.support.v7.widget.RecyclerView$AdapterDataObservable.notifyItemRangeInserted(RecyclerView.java:11382)
        at android.support.v7.widget.RecyclerView$Adapter.notifyItemInserted(RecyclerView.java:6745)
        at io.github.luizgrp.sectionedrecyclerviewadapter.SectionedRecyclerViewAdapter.callSuperNotifyItemInserted(SectionedRecyclerViewAdapter.java:548)
        at io.github.luizgrp.sectionedrecyclerviewadapter.SectionedRecyclerViewAdapter.notifyHeaderInsertedInSection(SectionedRecyclerViewAdapter.java:815)
        at io.github.luizgrp.sectionedrecyclerviewadapter.SectionedRecyclerViewAdapterTest.notifyHeaderInsertedInSectionUsingSection_withAdapterWithManySections_callsSuperNotifyItemInserted(SectionedRecyclerViewAdapterTest.java:876)

### CODE REVIEW

File: library/src/test/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionedRecyclerViewAdapterTest.java [hunk 1]
```diff
@@ -841,6 +841,44 @@ public void notifyItemMovedInSectionUsingSection_withAdapterWithManySections_cal
         verify(spySectionedRecyclerViewAdapter).callSuperNotifyItemMoved(11, 15);
     }
 
+    @Test
+    public void notifyHeaderInsertedInSectionUsingTag_withAdapterWithManySections_callsSuperNotifyItemInserted() {
+        // Given
+        SectionedRecyclerViewAdapter spySectionedRecyclerViewAdapter = spy(SectionedRecyclerViewAdapter.class);
+        doNothing().when(spySectionedRecyclerViewAdapter).callSuperNotifyItemMoved(anyInt(), anyInt());
```
Comment (luizgrp) [inline library/src/test/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionedRecyclerViewAdapterTest.java]: it should be callSuperNotifyItemInserted

File: library/src/test/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionedRecyclerViewAdapterTest.java [hunk 2]
```diff
@@ -841,6 +841,44 @@ public void notifyItemMovedInSectionUsingSection_withAdapterWithManySections_cal
...
+
+        // Then
+        verify(spySectionedRecyclerViewAdapter).callSuperNotifyItemInserted(10);
+    }
+
+    @Test
+    public void notifyHeaderInsertedInSectionUsingSection_withAdapterWithManySections_callsSuperNotifyItemInserted() {
+        // Given
+        SectionedRecyclerViewAdapter spySectionedRecyclerViewAdapter = spy(SectionedRecyclerViewAdapter.class);
+        doNothing().when(spySectionedRecyclerViewAdapter).callSuperNotifyItemMoved(anyInt(), anyInt());
```
Comment (luizgrp) [inline library/src/test/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionedRecyclerViewAdapterTest.java]: same here

File: library/src/test/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionedRecyclerViewAdapterTest.java (hunk 1 above)
Comment (yccheok) [inline library/src/test/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionedRecyclerViewAdapterTest.java]: Oh my. What a shame on my copy and paste without going through the code carefully. Thanks.

**Comment by luizgrp** [comment]
could you please pull develop and merge into your branch so we only see your changes? cheers

**Comment by yccheok** [comment]
x

Hi @luizgrp 

I created a branch from your master branch, and develop from there.

Am I doing correctly?

I believe the reason we are still seeing Frieder Bluemle, is because Frieder Bluemle change is in master branch, but not in develop branch?

**Comment by luizgrp** [comment]
You should have created your branch from develop.
I moved the changes from Frieder to develop then they will disappear from here once you get the latest changes from develop.

**Comment by yccheok** [comment]
ok. I created another pull request, which branched out from `develop` branch.

https://github.com/luizgrp/SectionedRecyclerViewAdapter/pull/65/files

**Pull request closed without merging**